ivy/functional/frontends/tensorflow/nn.py

```diff
@@ -145,6 +145,59 @@ def conv3d_transpose(
     )
 
 
+@with_unsupported_dtypes({"2.12.0 and below": ("bfloat16",)}, "tensorflow")
+@to_ivy_arrays_and_back
+def conv_transpose(
+    input,
+    filters,
+    output_shape,
+    strides,
+    padding="SAME",
+    data_format="NDHWC",
+    dilations=None,
+    name=None,
+):
+    N = len(input)-2
```

Comment:
This seems wrong:
```python
>>> arr = tf.constant([[1]])
>>> arr.shape
TensorShape([1, 1])
>>> len(arr)
1
```
`arr` is 2 dimensional here.

Reply:
I think the `len()` is the right way to do it because you put an array inside an empty array (basically a double array). The double array in this case doesn't make any point for us. In the function, we are trying to find how many dimensions of the first order are there, and in this case, there is only one dimension of the first order and not 2. If we are trying to figure out the "NDHWC", we are trying to figure out the dimension of the first order and we don't care if it is "NDHWC" or 
"[N]DHWC" or "[[N]]D[H]WC". We just need to know that there are 5 first-order dimensions. The other advantage of `len()` is that some people might still use a list data type, which would crash when using `shape()` on it. If the argument is restricted to arrays only, then Yes, you are right and we could use the `shape()` function.

Reply:
Now the `shape()` is also not so perfect, if we are trying to input something like this 
`arr = tf.constant([[[1, 2], [1, 2], [1, 2], [2, 2]]])`
`print(arr.shape)`
`print(len(arr))`
the output is 
`(1, 4, 2)`
`1`
and if you would index on the first element of the tuple to figure out the dimension you would get the wrong thing as you are expecting to count the first order dimentions.

Reply:
On the other hand if to remove one dimension and put
arr = tf.constant([[1, 2], [1, 2], [1, 2], [2, 2]])
print(arr.shape)
print(len(arr))
the output is
(4, 2)
4
which perfectly works with `len` too

Reply:
I don't think `N` is literally the first `N` dimension in `NC` or `NWC` or `NHWC`.
> input | An N+2 dimensional Tensor of shape [batch_size] + input_spatial_shape + [in_channels] if data_format does not start with "NC" (default), or [batch_size, in_channels] + input_spatial_shape if data_format starts with "NC"

The `N` you are describing is `batch_size`, and the `N` that is used throughout the docs (`N+2 dimensional Tensor`) is the number of dimensions of `input` minus 2. You can try this example:
```python
>>> arr = tf.random.uniform([1,3,224,224])
>>> arr.shape
TensorShape([1, 3, 224, 224])
# a single RGB image of size 224x224
# batch_size - channel - height - width
# so N is 2, and the correct format is either `NHWC` or `NCHW`
>>> len(arr)
1
# if N is 1, what is the correct format?
```

Reply:
Alright, so can I use the `shape` function? (then it will not work for lists). That's the reason I wanted to use `len()`. If I can use `shape` function I can just do `len(arr.shape)-2` which would give me the right number of dimensions. Another solution would be to transform all lists in numpy arrays before taking the `shape`.

Reply:
Does `tf.nn.conv_transpose` accept lists as input?